pkg/controller/keycloakbackup/keycloakbackup_controller.go

```diff
@@ -0,0 +1,170 @@
+package keycloakbackup
+
+import (
+	"context"
+	"time"
+
+	kc "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	"github.com/keycloak/keycloak-operator/pkg/common"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	RequeueDelaySeconds      = 30
+	RequeueDelayErrorSeconds = 5
+	ControllerName           = "keycloakbackup-controller"
+)
+
+var log = logf.Log.WithName("controller_keycloakbackup")
+
+func Add(mgr manager.Manager, _ chan schema.GroupVersionKind) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+
+	return &ReconcileKeycloakBackup{
+		client:   mgr.GetClient(),
+		scheme:   mgr.GetScheme(),
+		cancel:   cancel,
+		context:  ctx,
+		recorder: mgr.GetRecorder(ControllerName),
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("keycloakbackup-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource KeycloakBackup
+	err = c.Watch(&source.Kind{Type: &kc.KeycloakBackup{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner KeycloakBackup
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
```

Comment:
Should be watching for all secondary resources here. There is a Job, CronJob and PVC.

Reply:
Fixed.